pkg/core/metadata.go

```diff
@@ -19,8 +19,36 @@ func SpecMetadata(name string) string {
 	return numaryNamespace + name
 }
 
+type RawMetadata map[string]any
```

Comment:
I don't understand why we need a new struct here? I feel like we should try to use the same `Metadata`.
Why not change it to `type Metadata map[string]any`?